### PR TITLE
New discovery protocol over HTTPS

### PR DESCRIPTION
**Pull request by calmh**
Maybe have a look-see here as well

### CODE REVIEW

File: querysrv.go [hunk 1]
```diff
@@ -3,213 +3,305 @@
...
+		globalStats.Error()
+		http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
 	}
 }
 
-func (s *querysrv) Stop() {
-	panic("stop unimplemented")
-}
+func (s *querysrv) handleGET(w http.ResponseWriter, req *http.Request) {
+	if req.TLS == nil {
```
Comment (AudriusButkevicius) [inline querysrv.go]: This is impossible, but I guess we can't be too safe.

Comment (AudriusButkevicius) [inline querysrv.go]: We don't really need this for gets

File: querysrv.go [hunk 2]
```diff
@@ -3,213 +3,305 @@
...
+		globalStats.Error()
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return
 	}
 
-	return tx.Commit()
+	globalStats.Announce()
+
+	// TODO: Slowly increase this for stable clients
+	w.Header().Set("Reannounce-After", "1800")
```
Comment (AudriusButkevicius) [inline querysrv.go]: I assume the clients don't yet respect that?

**Comment by AudriusButkevicius** [comment]
Does this work?

File: querysrv.go (hunk 2 above)
Comment (calmh) [inline querysrv.go]: They do.

File: querysrv.go (hunk 1 above)
Comment (calmh) [inline querysrv.go]: Well, we don't even listen on non-TLS so not really, no.

**Comment by calmh** [comment]
Yeah :)

File: querysrv.go (hunk 1 above)
Comment (AudriusButkevicius) [inline querysrv.go]: This talks about wether or not the client provided a cert. I'd like to be able to easily ask people to use curl to debug stuff, without asking them to generate a key pair.

Comment (calmh) [inline querysrv.go]: Nah, this just verifies that the GET was over TLS at all, no check of the client cert. `curl -k` works perfectly fine against the discosrv for GET requests.

**Pull request merged**